doc/api/errors.markdown

```diff
@@ -0,0 +1,470 @@
+# Errors
+
+<!--type=misc-->
+
+Errors generated by io.js fall into two categories: JavaScript errors and system
+errors. All errors inherit from or are instances of JavaScript's [Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error)
+class and are guaranteed to provide *at least* the attributes available on that
+class.
+
+When an operation is not permitted due to language-syntax or
+language-runtime-level reasons, a **JavaScript error** is generated and thrown
+as an **exception**. If an operation is not allowed due to system-level
+restrictions, a **system error** is generated. Client code is then given the
+opportunity to **intercept** this error based on how the API **propagates** it.
+
+The style of API called determines how generated errors are handed back, or
```

Comment:
This first line should be obvious to readers, I don't think it needs to be said.